STL_Extension/include/CGAL/Handle_for.h

```diff
@@ -63,7 +81,11 @@ class Handle_for
     {
         pointer p = allocator.allocate(1);
         new (&(p->t)) element_type(); // we get the warning here
+#ifdef CGAL_HANDLE_FOR_USE_BOOST_ATOMIC_COUNTER
+        new (&(p->count)) boost::detail::atomic_count(1);
+#else // not Boost atomic counter
         p->count = 1;
+#endif // not Boost atomic counter
```

Comment:
Should not `RefCounted` have proper constructors, that would call the two placement new?